trunk/SUAVE/Components/Energy/Networks/Battery_Propeller.py

```diff
@@ -90,151 +98,263 @@ def evaluate_thrust(self,state):
+            
+        D            = numerics.time.differentiate        
+        battery_data = battery.discharge_performance_map  
         
         # Set battery energy
-        battery.current_energy = conditions.propulsion.battery_energy  
-
-        # Step 1 battery power
-        esc.inputs.voltagein = state.unknowns.battery_voltage_under_load
-        
-        # Step 2
-        esc.voltageout(conditions)
+        battery.current_energy      = conditions.propulsion.battery_energy
+        battery.pack_temperature    = conditions.propulsion.battery_pack_temperature
+        battery.charge_throughput   = conditions.propulsion.battery_charge_throughput     
+        battery.age_in_days         = conditions.propulsion.battery_age_in_days 
+        discharge_flag              = conditions.propulsion.battery_discharge    
+        battery.R_growth_factor     = conditions.propulsion.battery_resistance_growth_factor
+        battery.E_growth_factor     = conditions.propulsion.battery_capacity_fade_factor 
+        battery.max_energy          = conditions.propulsion.battery_max_aged_energy
+        V_th0                       = conditions.propulsion.battery_initial_thevenin_voltage
+        n_series                    = battery.pack_config.series  
+        n_parallel                  = battery.pack_config.parallel
+        n_total                     = n_series*n_parallel
         
-        # How many evaluations to do
-        if self.identical_propellers:
-            n_evals = 1
-            factor  = num_engines*1
-        else:
-            n_evals = int(num_engines)
-            factor  = 1.
-        
-        # Setup numbers for iteration
-        total_motor_current = 0.
-        total_thrust        = 0. * state.ones_row(3)
-        total_power         = 0.
+        # update ambient temperature based on altitude
+        battery.ambient_temperature                   = conditions.freestream.temperature   
+        battery.cooling_fluid.thermal_conductivity    = conditions.freestream.thermal_conductivity
+        battery.cooling_fluid.kinematic_viscosity     = conditions.freestream.kinematic_viscosity
+        battery.cooling_fluid.density                 = conditions.freestream.density  
+        battery.ambient_pressure                      = conditions.freestream.pressure  
+        a                                             = conditions.freestream.speed_of_sound
         
-        # Iterate over motor/props
-        for ii in range(n_evals):
-            
-            # Unpack the motor and props
-            motor_key = list(motors.keys())[ii]
-            prop_key  = list(props.keys())[ii]
-            motor     = self.propeller_motors[motor_key]
-            prop      = self.propellers[prop_key]
+        # Set battery energy
+        battery.current_energy = conditions.propulsion.battery_energy  
+
+
+        # --------------------------------------------------------------------------------
```

Comment:
Again, let's make this call the battery model instead of doing it here

Reply:
addressed, now packaged in the battery model itself